test/Dialect/FIRRTL/lowerXMR.mlir

```diff
@@ -117,3 +117,136 @@ firrtl.circuit "Top" {
+  }
+}
+
+// -----
+
+// Test for multiple paths and downward reference.
+// CHECK-LABEL: firrtl.circuit "Top" {
+firrtl.circuit "Top" {
+  firrtl.module @XmrSrcMod(out %_a: !firrtl.ref<uint<1>>) {
+    %zero = firrtl.constant 0 : !firrtl.uint<1>
+    %1 = firrtl.ref.send %zero : !firrtl.uint<1>
+    firrtl.strictconnect %_a, %1 : !firrtl.ref<uint<1>>
+  }
+  firrtl.module @Foo(out %_a: !firrtl.ref<uint<1>>) {
+    %xmr   = firrtl.instance bar sym @fooXMR @XmrSrcMod(out _a: !firrtl.ref<uint<1>>)
+    firrtl.strictconnect %_a, %xmr   : !firrtl.ref<uint<1>>
+  }
+  firrtl.module @Top() {
+    %foo_a = firrtl.instance foo sym @foo @Foo(out _a: !firrtl.ref<uint<1>>)
+    %xmr_a = firrtl.instance xmr sym @xmr @XmrSrcMod(out _a: !firrtl.ref<uint<1>>)
+    %c_a, %c_b = firrtl.instance child @Child2p(in _a: !firrtl.ref<uint<1>>, in _b: !firrtl.ref<uint<1>> )
+    // CHECK:  firrtl.instance child  @Child2p()
+    firrtl.strictconnect %c_a, %foo_a : !firrtl.ref<uint<1>>
+    firrtl.strictconnect %c_b, %xmr_a : !firrtl.ref<uint<1>>
+  }
+  firrtl.module @Child2p(in  %_a: !firrtl.ref<uint<1>>, in  %_b: !firrtl.ref<uint<1>>) {
+    %0 = firrtl.ref.resolve %_a : !firrtl.ref<uint<1>>
+    // CHECK-LITERAL:  %1 = firrtl.verbatim.expr "{{0}}.{{1}}.{{2}}" : () -> !firrtl.uint<1> {symbols = [#hw.innerNameRef<@Top::@foo>, #hw.innerNameRef<@Foo::@fooXMR>, #hw.innerNameRef<@XmrSrcMod::@xmr_sym>]}
+    %1 = firrtl.ref.resolve %_b : !firrtl.ref<uint<1>>
+    // CHECK-LITERAL:  %1 = firrtl.verbatim.expr "{{0}}.{{1}}" : () -> !firrtl.uint<1> {symbols = [#hw.innerNameRef<@Top::@xmr>, #hw.innerNameRef<@XmrSrcMod::@xmr_sym>]}
+  }
+}
+
+
+// -----
+
+// Test for multiple children paths
+// CHECK-LABEL: firrtl.circuit "Top" {
+firrtl.circuit "Top" {
+  firrtl.module @XmrSrcMod(out %_a: !firrtl.ref<uint<1>>) {
+    %zero = firrtl.constant 0 : !firrtl.uint<1>
+    %1 = firrtl.ref.send %zero : !firrtl.uint<1>
+    firrtl.strictconnect %_a, %1 : !firrtl.ref<uint<1>>
+  }
+  firrtl.module @Top() {
+    %xmr_a = firrtl.instance xmr sym @xmr @XmrSrcMod(out _a: !firrtl.ref<uint<1>>)
+    %c_a = firrtl.instance child @Child1(in _a: !firrtl.ref<uint<1>>)
+    firrtl.strictconnect %c_a, %xmr_a : !firrtl.ref<uint<1>>
+  }
+  // CHECK-LABEL: firrtl.module @Child1() {
+  firrtl.module @Child1(in  %_a: !firrtl.ref<uint<1>>) {
+    %0 = firrtl.ref.resolve %_a : !firrtl.ref<uint<1>>
+    // CHECK-LITERAL:  firrtl.verbatim.expr "{{0}}.{{1}}" : () -> !firrtl.uint<1> {symbols = [#hw.innerNameRef<@Top::@xmr>, #hw.innerNameRef<@XmrSrcMod::@xmr_sym>]}
+    %c_a, %c_b = firrtl.instance child @Child2(in _a: !firrtl.ref<uint<1>>, in _b: !firrtl.ref<uint<1>> )
+    firrtl.strictconnect %c_a, %_a : !firrtl.ref<uint<1>>
+    firrtl.strictconnect %c_b, %_a : !firrtl.ref<uint<1>>
+    %c3 = firrtl.instance child @Child3(in _a: !firrtl.ref<uint<1>>)
+    firrtl.strictconnect %c3 , %_a : !firrtl.ref<uint<1>>
+  }
+  firrtl.module @Child2(in  %_a: !firrtl.ref<uint<1>>, in  %_b: !firrtl.ref<uint<1>>) {
```

Comment:
Just curious, is `%_a` needed/useful here?

Reply:
I missed the `resolve` for it, but true, it might not be useful here.